pyERGM/metrics.py

```diff
@@ -1182,3 +1326,70 @@ def get_ignored_features(self):
             parameter_names += metric._get_ignored_features()
 
         return parameter_names
+
+    def bootstrap_observed_features(self, observed_network: np.ndarray, num_subsamples: int = 1000,
+                                    splitting_method: str = 'uniform'):
```

Comment:
won't splitting_method be determined by the metrics that exist in self.metrics?

Reply:
I didn't know how to deal with that properly (as all the future use cases are not clear). An obvious use case is that there is a clear best solution that the user may not be aware of (e.g. with types). In this case, there are multiple possible mechanisms to deal with user inputs (if there are) - override, raise error, warn etc.
I though that maybe we will have in the future cases where it is not clear which splitting method to choose, and then deciding for the user is weird (and maybe we can even allow to pass custom splitting methods if users want to).